### ISSUE TITLE
unable to call var args function at compile time

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#313]
```
test "runtime parameter before var args" {
    comptime {
        assert(extraFn(10) == 0);
        assert(extraFn(10, false) == 1);
        assert(extraFn(10, false, true) == 2);
    }
}

fn extraFn(extra: u32, args: ...) -> usize {
    if (args.len >= 1) {
        assert(args[0] == false);
    }
    if (args.len >= 2) {
        assert(args[1] == true);
    }
    return args.len;
}
```

**Comment by japplegame** [comment]
How difficult is it to fix this?

**Comment by emekoi** [comment]
this is being fixed by #208, which removes zig var args in favor of anonymous lists.

**Comment by andrewrk** [comment]
Here's what this looks like with tuples:

```zig
const std = @import("std");
const assert = std.debug.assert;

test "runtime parameter before var args" {
    comptime {
        assert(extraFn(10, .{}) == 0);
        assert(extraFn(10, .{false}) == 1);
        assert(extraFn(10, .{ false, true }) == 2);
    }
}

fn extraFn(extra: u32, args: var) usize {
    if (args.len >= 1) {
        assert(args[0] == false);
    }
    if (args.len >= 2) {
        assert(args[1] == true);
    }
    return args.len;
}
```

```
$ ./zig test test.zig 
All 1 tests passed.
```

var args removed from the language in a3f6a58c7785e7958f9d0b96d54356944bf34e32
